sdks/python/apache_beam/io/textio.py

```diff
@@ -0,0 +1,264 @@
+        # at current_pos.
+        if not self._try_to_ensure_num_bytes_in_buffer(current_pos + 1):
+          return
+
+      # Using find() here is more efficient than a linear scan of the byte
+      # array.
+      next_lf = self._buffer.find('\n', current_pos)
+      if next_lf >= 0:
+        if self._buffer[next_lf - 1] == '\r':
+          return (next_lf - 1, next_lf + 1)
+        else:
+          return (next_lf, next_lf + 1)
+
+      current_pos = len(self._buffer)
+
+  def _try_to_ensure_num_bytes_in_buffer(self, num_bytes):
+    # Tries to ensure that there are at least num_bytes bytes in the buffer.
+    # Returns True if this can be fulfilled, returned False if this cannot be
+    # fulfilled due to reaching EOF.
+    while len(self._buffer) < num_bytes:
+      read_data = self._file.read(self._buffer_size)
+      if not read_data:
+        return False
+
+      self._buffer += read_data
+
+    return True
+
+  def _read_record(self):
+    # Returns a tuple containing the current_record and number of bytes to the
+    # next record starting from 'self._next_position_in_buffer'. If EOF is
+    # reached, returns a tuple containing the current record and -1.
+
+    if self._next_position_in_buffer > self._buffer_size:
+      # Buffer is too large. Truncating it and adjusting
+      # self._next_position_in_buffer.
+      self._buffer = self._buffer[self._next_position_in_buffer:]
+      self._next_position_in_buffer = 0
+
+    record_start_position_in_buffer = self._next_position_in_buffer
+    sep_bounds = self._find_separator_bounds()
+    self._next_position_in_buffer = sep_bounds[1] if sep_bounds else len(
+        self._buffer)
+
+    if not sep_bounds:
+      # Reached EOF. Bytes up to the EOF is the next record. Returning '-1' for
+      # the starting position of the next record.
+      return (self._buffer[record_start_position_in_buffer:], -1)
+
+    if self._strip_trailing_newlines:
+      # Current record should not contain the separator.
+      return (self._buffer[record_start_position_in_buffer:sep_bounds[0]],
+              sep_bounds[1] - record_start_position_in_buffer)
+    else:
+      # Current record should contain the separator.
+      return (self._buffer[record_start_position_in_buffer:sep_bounds[1]],
+              sep_bounds[1] - record_start_position_in_buffer)
+
+
+class _TextSink(fileio.TextFileSink):
+  # TODO: Move code from 'fileio.TextFileSink' to here.
+  pass
+
+
+class ReadFromText(PTransform):
+  """A PTransform for reading text files.
+
+  Parses a text file as newline-delimited elements, by default assuming
+  UTF-8 encoding. Supports newline delimiters '\n' and '\r\n'.
+
+  This implementation only supports reading text encoded using UTF-8 or ASCII.
+  This does not support other encodings such as UTF-16 or UTF-32."""
+
+  def __init__(self, file_pattern=None, min_bundle_size=0,
+               compression_type=fileio.CompressionTypes.NO_COMPRESSION,
```

Comment:
What are all the places where we consult "splittable"?

A few places I sampled, do it when we have access to the sole (ie fully resolved) underlying file, in which case one could (also) consult ChannelFactory.is_compressed(file), or?

Would that allow us supporting AUTO immediately (ie as part of this CL)?
That would be preferred since otherwise there is a slight regression with this CL, since the NativeFileSource supports AUTO.